### PR TITLE
Fixes #34324 - Exclude infra permission from default roles

### PR DESCRIPTION
**Pull request by adamruzicka**
Requires:
- https://github.com/theforeman/foreman/pull/9064
- https://github.com/theforeman/foreman/pull/9073

### CODE REVIEW

**Comment by adamruzicka** [comment]
```
# for role in "Manager" "Site manager" "Organization admin"; do echo "$role"; h role filters --name "$role" | grep JobInvocation; echo; done

Manager
169 | JobInvocation                | none   | yes        | no        | Manager | create_job_invocations, view_job_invocations, execute_jobs_on_infrastructure_...

Site manager
145 | JobInvocation       | none   | yes        | no        | Site manager | create_job_invocations, view_job_invocations, execute_jobs_on_infrastructure_...

Organization admin
173 | JobInvocation                | none   | yes        | no        | Organization admin | create_job_invocations, view_job_invocations, cancel_job_invocations

```

**Comment by ezr-ondrej** [comment]
I've rekicked the job, tho if you want all of this CPed to foreman 3.1, I'll be a sad :panda_face: because we will have to implement the core changes completely anew, as the implementation changed quite significantly in 3.2

File: db/seeds.d/20-permissions.rb [hunk 1]
```diff
@@ -7,3 +7,15 @@
     filter.filterings.build { |f| f.permission = view_permission }
   end
 end
+
+site_manager_role = Role.find_by(name: 'Site manager')
```
Comment (ezr-ondrej) [inline db/seeds.d/20-permissions.rb:11]: Can't we just add the 'Site manager' to `AUTO_EXTENDED_ROLES` and edit it in the engine file?
 https://github.com/theforeman/foreman/blob/ef2ff06831ae7efcdab247b13c9ff642c6d56688/app/registries/foreman/plugin/rbac_support.rb#L5

Comment (adamruzicka) [reply db/seeds.d/20-permissions.rb:11]: We definitely could, do you know if there's a reason why it currently is excluded from AUTO_EXTENDED_ROLES?

**Comment by adamruzicka** [comment]
> I've rekicked the job, tho if you want all of this CPed to foreman 3.1, I'll be a sad panda_face because we will have to implement the core changes completely anew

Sadly yes. However looking at 3.1, it might be quite a simple change, unless I'm missing something.

File: db/seeds.d/20-permissions.rb (hunk 1 above)
Comment (adamruzicka) [reply db/seeds.d/20-permissions.rb]: Nevermind, did that

Comment (ezr-ondrej) [reply db/seeds.d/20-permissions.rb]: Because it was added later and noone actually thought of adding it :shrug:

**Comment by ezr-ondrej** [comment]
> Sadly yes. However looking at 3.1, it might be quite a simple change, unless I'm missing something.

Yeah, it won't be that bad, just that it will be all harder to CP ...

**Comment by adamruzicka** [comment]
Could we get this rolling again?

**Comment by ezr-ondrej** [comment]
Oh, I've missed this did not get in yet. Sorry :facepalm:

**Pull request merged**